### PR TITLE
Set maximum height to the popover to prevent cut-off on smaller screens

### LINKED ISSUE #960: Add types to `composables/use-popper.js`
## Description

Add type checking to `use-popper.js`. This module has zero untyped dependencies. Make sure to add it to `tsconfig.json`'s `include` list.

## Additional context

Part of an ongoing effort to add type checking to parts of the project that can be type checked.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### LINKED ISSUE #1212: Pages menu is cut off in mobile landscape view
## Description
Some of Menu items not visible.

## Reproduction
1. Visit: https://wordpress.org/openverse/
2. Search anything using images search option in search box.
3. Change the screen viewport to the mobile devices.
4. Click on Button(Dots) https://prnt.sc/HutwN50nGtmO
5. Menu items showing till "API"  then rest of items not visible.

## Screenshots
https://prnt.sc/aUuprxKjkp2m

## Environment
 - Device: iPhone 12 Pro Landscape mode
 - Browser: Chrome

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1212 by @Asadpalasara-multidots

Also fixes #960

## Description

This PR adds a `clippable` prop to `VPopover` that makes the popover only take up the available window height and scroll if the content does not fit.
Currently, when the popover content becomes scrollable, I can also see a scroll line in the bottom. I need to figure out how to remove it.

## Testing Instructions

Open a search results page, make the window height small so that the menu does not fit in. Then, click on the page menu or the content switcher. You should see that the popover is only as high as the window, and its content is scrollable:

clip-popover

The popover should also update if you change the window height.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by panchovm** [comment]
It looks great. Great improvement.

Is it possible to set a variable height? I am thinking of `100vh - 20px`, the view's height minus 20px. It looks good in my browser inspector.

**Review by panchovm — APPROVED** [review]
It was excellent. Great work ✨

File: src/components/VPopover/VPopover.vue [hunk 1]
```diff
@@ -128,7 +129,16 @@ export default defineComponent({
     /**
      * the z-index to apply to the popover content
      */
-    zIndex: { type: Number, default: 999 },
+    zIndex: {
+      type: Number,
+      default: 50,
+      validator: (v) => [0, 10, 20, 30, 40, 50].includes(v),
```
Comment (zackkrida) [inline src/components/VPopover/VPopover.vue:135]: Could we add a `todo` comment to get these z index layers in a type somewhere, if we plan to use these as a z-index system across the frontend?

**Review by zackkrida — APPROVED** [review]
LGTM

File: src/components/VPopover/VPopover.vue (hunk 1 above)
Comment (obulat) [reply src/components/VPopover/VPopover.vue:135]: I used the default tailwind values for z-index here: https://tailwindcss.com/docs/z-index
I guess I need to look into how I can import them from the Tailwind config object.

Comment (zackkrida) [reply src/components/VPopover/VPopover.vue:135]: oh, i see. i think it's fine then. if i remember correctly the config importing is something like this:

```js
import resolveConfig from 'tailwindcss/resolveConfig'
import tailwindConfig from './tailwind.config.js'

const tailwindconfig = resolveConfig(tailwindConfig)
```

Comment (obulat) [reply src/components/VPopover/VPopover.vue:136]: I added a todo comment. We need the array of valid values in several places (at least in VPopover and VPopoverContent), so it's a bigger change than I expected.

Comment (zackkrida) [reply src/components/VPopover/VPopover.vue:136]: cool, a standalone issue for that would be great. definitely doesn't need to happen here. LGTM!

Comment (zackkrida) [reply src/components/VPopover/VPopover.vue:136]: I made one! https://github.com/WordPress/openverse-frontend/issues/1742

**Review by zackkrida — APPROVED** [review]

**Review by dhruvkb — APPROVED** [review]

**Pull request merged**